### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by Nereuxofficial**
These changes are sometimes simplifications sometimes unnecessary references. None of them change the functionality of anything but are mostly just easier to read or slightly increase Performance(like using Mutex::get_mut instead of Mutex::lock).

### CODE REVIEW

File: src/topology/gene.rs [hunk 1]
```diff
@@ -239,9 +239,9 @@ where
     T: Float,
 {
     fn cmp(&self, other: &Self) -> Ordering {
-        if self.output.layer == other.output.layer && self.output.index == other.output.index {
+        if self.output.layer == other.output.layer {
```
Comment (sakex) [inline src/topology/gene.rs:242]: This should stay the same `self.output.layer == other.output.layer && self.output.index == other.output.index`

File: src/topology/gene.rs [hunk 2]
```diff
@@ -239,9 +239,9 @@ where
...
             Ordering::Equal
-        } else if self.output.layer < other.output.layer && self.output.layer < other.output.layer {
+        } else if self.output.layer < other.output.layer {
```
Comment (sakex) [inline src/topology/gene.rs:244]: This one should be

`else if self.output.layer < other.output.layer  || (self.output.layer == other.output.layer && self.output.layer < other.output.layer)`

File: src/topology/mutation_probabilities.rs [hunk 3]
```diff
@@ -35,17 +35,15 @@ impl MutationProbabilities {
         change_weights: f64,
         guaranteed_new_neuron: f64,
     ) -> Result<MutationProbabilities, ProbabilitiesError> {
-        if change_weights > 1.0
-            || change_weights < 0.0
-            || guaranteed_new_neuron > 1.0
-            || guaranteed_new_neuron < 0.0
-        {
+        let range = 0.0..1.0;
+        if !range.contains(&guaranteed_new_neuron) || !range.contains(&change_weights) {
```
Comment (sakex) [inline src/topology/mutation_probabilities.rs:39]: Thanks I didn't know about that, it looks good

**Review by sakex — CHANGES_REQUESTED** [review]
Thanks for your review, I really appreciate it! I just had a minor comment, otherwise LGTM

**Comment by Nereuxofficial** [comment]
Thanks for the feedback! I implemented the changes you requested. 
I also updated the dependencies which didn't require further changes.

**Review by sakex — APPROVED** [review]

**Comment by sakex** [comment]
@Nereuxofficial Thanks, it's merged!

**Comment by sakex** [comment]
@Nereuxofficial I added you as collaborator so you don't have to fork every time

**Comment by sakex** [comment]
@Nereuxofficial  I also added CI so tests are run before merging

**Comment by Nereuxofficial** [comment]
Thanks! I sadly don't have much experience yet with NEAT or GRUs but i'm reading into it.

**Pull request merged**